pallets/controller/src/lib.rs

```diff
@@ -480,17 +468,16 @@ impl<T: Config> Pallet<T> {
 		redeem_amount: Balance,
 		borrow_amount: Balance,
 	) -> LiquidityResult {
-		let m_tokens_ids: Vec<CurrencyId> = <T as liquidity_pools::Config>::EnabledCurrencyPair::get()
-			.iter()
-			.map(|currency_pair| currency_pair.wrapped_id)
-			.collect();
+		let m_tokens_ids: Vec<CurrencyId> = CurrencyId::get_enabled_tokens_in_protocol(WrappedToken);
 
 		let mut sum_collateral = Balance::zero();
 		let mut sum_borrow_plus_effects = Balance::zero();
 
 		// For each tokens the account is in
 		for asset in m_tokens_ids.into_iter() {
-			let underlying_asset = <LiquidityPools<T>>::get_underlying_asset_id_by_wrapped_id(&asset)?;
+			let underlying_asset = asset
+				.get_underlying_asset_id_by_wrapped_id()
```

Comment:
Since you no longer operate with ids here, the function should be renamed. What do you think?
```suggestion
				.get_underlying_asset_id()
```

Reply:
Sounds good. Fixed.